web/src/main/java/com/microsoft/applicationinsights/web/internal/cookies/SessionCookie.java

```diff
@@ -103,99 +72,18 @@ public String getSessionId() {
-                this.getSessionRenewalDate(),
-                Calendar.MINUTE,
-                sessionTimeoutInMinutes);
-        Date now = new Date();
-
-        return  now.after(expirationDate);
-    }
-
-    /**
-     * Returns a value indicating whether the session cookie is up-to-date.
-     * Session cookie is considered up-to-date when the last renewal time is less than
-     * {@literal UPDATE_TIMEOUT_IN_MINUTES} minutes.
-     * @return True if the session cookie up-to-date.
-     */
-    public boolean isSessionCookieUpToDate() {
-        Date expectedRenewalTime = DateTimeUtils.addToDate(
-                this.getSessionRenewalDate(),
-                Calendar.MINUTE,
-                UPDATE_TIMEOUT_IN_MINUTES);
-        Date now = new Date();
-
-        return now.before(expectedRenewalTime);
-    }
-
     // endregion Public
 
     // region Private
 
     /**
      * Parses the given cookie.
      * @param cookie The cookie contains the session information.
-     * @throws Exception Thrown when the cookie information cannot be parsed.
+     * @return sessionId 
      */
-    private void parseCookie(Cookie cookie) throws Exception {
-        String[] split = cookie.getValue().split(RAW_COOKIE_SPLIT_DELIMITER);
-
-        if (split.length < CookieFields.values().length) {
-
-            // TODO: dedicated exception
-            String errorMessage = String.format("Session cookie is not in the correct format: %s", cookie.getValue());
-
-            throw new Exception(errorMessage);
-        }
-
-        try {
-            sessionId = split[CookieFields.SESSION_ID.getValue()];
-            acquisitionDate = parseDateWithBackwardCompatibility(split[CookieFields.SESSION_ACQUISITION_DATE.getValue()]);
-            renewalDate = parseDateWithBackwardCompatibility(split[CookieFields.SESSION_LAST_UPDATE_DATE.getValue()]);
-        } catch (Exception e) {
-            String errorMessage = String.format("Failed to parse session cookie with exception: %s", e.getMessage());
-
-            // TODO: dedicated exception
-            throw new Exception(errorMessage);
-        }
-    }
-
-    /**
-     * JavaScript SDK was changed to store dates as long, rather than a readable date string.
-     * For backward compatibility, we first try to parse with the new format (time represented by long) and then backward
-     * compatibility for time represented by a string.
-     * @param dateStr The date to parse.
-     * @return The parsed date.
-     */
-    private Date parseDateWithBackwardCompatibility(String dateStr) throws ParseException {
-        try {
-            return new Date(Long.parseLong(dateStr));
-        } catch (NumberFormatException e) {
-            return DateTimeUtils.parseRoundTripDateString(dateStr);
-        }
+    private static String parseCookie(Cookie cookie) {
+        String[] split = cookie.getValue().split(RAW_COOKIE_SPLIT_DELIMITER, 2);
+        return split[0];
```

Comment:
substring will work faster

Reply:
I agree 
I will change it to use substring